ledger/controllers/surveyor.js

```diff
@@ -708,3 +739,24 @@ module.exports.initialize = async (debug, runtime) => {
     }
   }
 }
+
+function getVoteRate () {
+  return (runtime) => async (request, reply) => {
+    const {
+      payload
+    } = await server(request, reply, runtime)
+    const {
+      adFree
+    } = payload
+
+    const bigProbi = new BigNumber(adFree.probi)
+    const bigVotes = new BigNumber(adFree.votes)
+    const voteRateProbi = bigProbi.dividedBy(bigVotes)
+    const voteRateBat = voteRateProbi.dividedBy(1e18)
+    const voteRate = voteRateBat.toString()
+
```

Comment:
It looks like this code is doing something similar as https://github.com/brave-intl/bat-ledger/blob/f82076649ad124e121e2d72a5d9b88d0f7ac4ef5/ledger/controllers/wallet.js#L218-L222

Do you think there's a way to consolidate this logic?

Reply:
yeah that makes sense